web/packages/teleterm/src/ui/services/workspacesService/documentsService/documentsService.ts

```diff
@@ -368,6 +374,45 @@ export class DocumentsService {
     });
   }
 
+  refreshPtyTitle(
+    uri: DocumentUri,
+    {
+      shell,
+      cwd,
+      clusterName,
+      runtimeSettings,
+    }: {
+      shell: Shell;
+      cwd: string;
+      clusterName: string;
+      runtimeSettings: Pick<RuntimeSettings, 'platform' | 'defaultOsShellId'>;
+    }
+  ) {
+    const doc = this.getDocument(uri);
+    if (!doc) {
+      return;
+    }
```

Comment:
Is it expected that `refreshPtyTitle` is called with a URI that no longer exists?

Reply:
No, removed.

Reply:
I don't think it was a bad idea to check if `doc` exists, it just that silently returning from the function could result in some callsite being broken. We wouldn't know about it until we reproduced the issue ourselves and saw that in some cases `refreshPtyTitle` is called with a URI that doesn't exist.

Throwing an error would be an acceptable way to deal with this I feel. At the moment if `doc` doesn't exist for whatever reason, we're going to get an error about trying to read `doc.kind`.